lib/concurrent/atomic/read_write_lock.rb

```diff
@@ -192,7 +203,8 @@ def acquire_write_lock
     #
     # @return [Boolean] true if the lock is successfully released
     def release_write_lock
-      c = @Counter.update { |c| c-RUNNING_WRITER }
+      c = @Counter.update { |count| count-RUNNING_WRITER }
+      @WriterLock.set(nil)
```

Comment:
Good catch, @ianks! What you say is absolutely right.

(Your first comment is an example of what I said about this PR: the name `WriterLock` is confusing. Something like `WriteLockHolder` would be more accurate.)